### PR TITLE
⬆️ Update Base image to v.12.0.0

### PR DESCRIPTION
**Pull request by elcajon**
# Proposed Changes

Base image version updated with breaking changes included. s6-overlay was updated from v2 to latest v3.
See https://github.com/hassio-addons/addon-base/releases/tag/v12.0.0

I updated files and structures to work with latest s6 version.
The new service definition is quite more complex than the legacy /etc/cont-init.d and /etc/services.d directives. 

## Related Issues

> ([Github link][autolink-references] to related issues or pull requests)

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

### CODE REVIEW

**Comment by brenner-tobias** [comment]
Thanks a lot for bringing that up. 
I think, we also have to update the apparmor file (see [here](https://developers.home-assistant.io/blog/2022/05/12/s6-overlay-base-images/#apparmor)), don't we? this would also be a great time to check it properly, since I had no idea when creating it, to be honest...

**Review by brenner-tobias — CHANGES_REQUESTED** [review]
Great, thanks. I think we should also look into the Apparmor file.

File: cloudflared/rootfs/build.sh [hunk 1]
```diff
@@ -40,4 +40,4 @@ chmod +x /usr/bin/cloudflared
 
 # Add needed packages
 apk add --no-cache \
-    yq=4.14.1-r2
```
[the remark this answers is not in the document]
Comment (elcajon) [reply cloudflared/rootfs/build.sh:43]: We can separate that but we need this version anyway as the new base image also updates alpine from 3.15. to 3.16. which invalidates the old yq version.

**Comment by elcajon** [comment]
> Thanks a lot for bringing that up.
> I think, we also have to update the apparmor file (see [here](https://developers.home-assistant.io/blog/2022/05/12/s6-overlay-base-images/#apparmor)), don't we? this would also be a great time to check it properly, since I had no idea when creating it, to be honest...

You are right, I hoped you can have a look at it as I have no clue about Apparmor neither. 😄

**Comment by elcajon** [comment]
I just updated the apparmor file paying attention to [this example](https://developers.home-assistant.io/docs/add-ons/presentation#apparmor).
I have no clue if I missed something in this part:

```
[...]
    # Access required for service functionality
    /usr/bin/cloudflared r,
[...]
```

I guess we should merge it and test the edge version.

File: cloudflared/rootfs/build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (brenner-tobias) [reply cloudflared/rootfs/build.sh:43]: Ah ok, makes sense then to just leave it

**Review by brenner-tobias — APPROVED** [review]

**Comment by brenner-tobias** [comment]
So I did merge it in and updated my HA at home. Now unfortunately, I cannot connect anymore and since I am not at home, I have no idea, why :-)
Is this working for you @elcajon?
I will be back home tomorrow and check, what's going on there.

**Pull request merged**